src/main/java/org/kuali/kfs/module/ar/document/web/struts/CashControlAction.java

```diff
@@ -0,0 +1,443 @@
+        final CashControlDetail cashControlDetail = cashControlDocument.getCashControlDetail(indexOfLineToDelete);
+        cancelLinkedPaymentApplicationDocument(cashControlDocument, cashControlDetail);
+
+        // load document so the displayed total amount is accurate and pay app is displayed as canceled
+        loadDocument(cashControlDocForm);
+        return mapping.findForward(KFSConstants.MAPPING_BASIC);
+    }
+
+    /**
+     * Recalculates the cash control total since user could have changed it during their update.
+     *
+     * @param cashControlDocument
+     */
+    protected KualiDecimal calculateCashControlTotal(final CashControlDocument cashControlDocument) {
+        KualiDecimal total = KualiDecimal.ZERO;
+        for (final CashControlDetail cashControlDetail : cashControlDocument.getCashControlDetails()) {
+            total = total.add(cashControlDetail.getFinancialDocumentLineAmount());
+        }
+        return total;
+    }
+
+    /**
+     * Processes a PaymentApplicationDocument linked to a given cash control to be deleted. Documents will only be
+     * processed if they are not already cancelled or disapproved. Enroute docs are disapproved, all others are cancelled.
+     * The provided cash control detail will also be cancelled and its amount will be removed from the cash control
+     * document's total amount
+     *
+     * @param cashControlDocument The cash control document that the provided cash control detail belongs to
+     * @param cashControlDetail The cash control detail to be canceled
+     */
+    private void cancelLinkedPaymentApplicationDocument(
+            final CashControlDocument cashControlDocument,
+            final CashControlDetail cashControlDetail
+    ) {
+        final DocumentService documentService = SpringContext.getBean(DocumentService.class);
+        final PaymentApplicationDocument payAppDoc = (PaymentApplicationDocument) documentService
+                .getByDocumentHeaderId(cashControlDetail.getReferenceFinancialDocumentNumber());
+        final WorkflowDocument workflowDocument = payAppDoc.getDocumentHeader().getWorkflowDocument();
+
+        if (!workflowDocument.isCanceled() && !workflowDocument.isDisapproved()) {
+            if (workflowDocument.isEnroute()) {
+                documentService.superUserDisapproveDocument(payAppDoc, ArKeyConstants.DOCUMENT_DELETED_FROM_CASH_CTRL_DOC);
+            } else {
+                documentService.cancelDocument(payAppDoc, ArKeyConstants.DOCUMENT_DELETED_FROM_CASH_CTRL_DOC);
+            }
+        }
+        cashControlDocument.recalculateTotals();
+    }
+
+    /**
+     * Import items to the document from a spreadsheet.
+     *
+     * @param mapping  An ActionMapping
+     * @param form     An ActionForm
+     * @param request  The HttpServletRequest
+     * @param response The HttpServletResponse
+     * @return An ActionForward
+     */
+    public ActionForward importDetails(
```

Comment:
The functions added  should also have a comment that they are part of the backport